### DISCUSSION TITLE
Searche/Favorite/Options is inconsistent, why?

### DISCUSSION THREAD

**Discussion by jpkanter** [vufind-org/vufind#4084] [General]
While working on an issue today I stumbled upon a weird inconsistency in regards of favorites:

If one, for any reasons, wants to adjusts the number for displayed favorites on the myresearch page or add a control widget for the amount of entries per page ones needs to add a new section to the main config named `[Social]` and the parameters are then named `lists_default_limit`and `lists_limit_options` this is unlike the "real" search options that are named `default_limit` and `limit_options`

I only found this after delving deeply into the code, namely `Search/Favorites/Options.php` to find the offending text:

```php
if (isset($config->Social->lists_default_limit)) {
    $this->defaultLimit = $config->Social->lists_default_limit;
}
if (isset($config->Social->lists_limit_options)) {
    $this->limitOptions = $this->explodeListSetting($config->Social->lists_limit_options);
}
```

According to codeblame this is almost a decade old code (last edited 2016-04-25 22:09 GMT+1), I did not wanted to open a pull request from the get go as this is a breaking change for all instances that modify this parameter. Also I put in zero research to see where else this parameter is used.

If its nowhere I would propose moving this parameter to the `searches.ini` under either `[Favorites]` or if broader used `[MyResearch]` and the properly named as the normal parameters that govern this feature for the search.

**Comment by demiankatz** [comment]
@jpkanter, the `[Social]` section is included in config.ini by default, and these settings are documented there. See:

https://github.com/vufind-org/vufind/blob/dev/config/vufind/config.ini#L2552-L2555

The reason for the inconsistency is that favorites are a social feature that have a broader scope than a search backend. Items in favorites can come from any backend. So the limit settings in searches.ini apply to Solr, the limit settings in EDS.ini apply to EDS, etc., etc. Favorites apply to EVERYTHING, and so these settings are in the universal primary configuration file, config.ini.

Does that help?

**Comment by jpkanter** [comment]
Yes, it does. I can see now why the got the `list_` prefix.
